error.go

```diff
@@ -0,0 +1,67 @@
+package mustache
+
+import (
+	"fmt"
+)
+
+// ErrorCode is the list of allowed values for the error's code.
+type ErrorCode string
+
+// List of values that ErrorCode can take.
+const (
+	ErrUnmatchedOpenTag      ErrorCode = "unmatched_open_tag"
+	ErrEmptyTag              ErrorCode = "empty_tag"
+	ErrSectionNoClosingTag   ErrorCode = "section_no_closing_tag"
+	ErrInterleavedClosingTag ErrorCode = "interleaved_closing_tag"
+	ErrInvalidMetaTag        ErrorCode = "invalid_meta_tag"
+	ErrUnmatchedCloseTag     ErrorCode = "unmatched_close_tag"
+)
+
+// Error represents an error during the parsing
+type Error struct {
+	// Line returns the line of the error
+	Line int
+	// Code returns the error code of the error
+	Code ErrorCode
+	// returns the name of the element generating the error
```

Comment:
Should be:
```go
// Reason contains the name of the element generating the error (if any)
```
Let's rename `returns` -> `contains` in the descriptions of `Line` and `Code` as well.

Reply:
I fixed the documentation